integr8ly/cloud-resource-operator
# [MGDAPI-3827] Tag redis & postgres snapshots on creation

## Overview

Jira: https://issues.redhat.com/browse/MGDAPI-3827

## Verification

- Provision a ROSA cluster using the script in delorean: `make -f make/ocm.mk ocm/sts/cluster/create OCM_ENV=production CLUSTER_NAME=sts-test`
- Setup and wait for cluster admin credentials to become available`rosa create admin --cluster=sts-test`
- Log into the cluster `oc login $(cluster_url) --username cluster-admin --password $(password)`
- Prepare the cluster `make cluster/prepare` using the make commands from the CRO repo
- Setup the sts secret `make setup/sts OCM_ENV=production CLUSTER_NAME=sts-test`
- Run CRO `make run`
- Deploy a redis instance`make cluster/seed/managed/redis`
- Deploy a postgres instance `make cluster/seed/managed/postgres`
- Wait for successful reconciliation
- Create a redis snapshot `make cluster/seed/managed/redissnapshot`
- Create a postgres snapshot `make cluster/seed/managed/postgressnapshot`
- Wait for successful reconciliation
- Ensure the redis and postgres snapshots have been tagged with `red-hat-managed` and other tags
- Trigger deletion of the redis snapshot `oc delete redissnapshot example-redissnapshot -n cloud-resource-operator`
- Trigger deletion of the postgres snapshot `oc delete postgressnapshot example-postgressnapshot -n cloud-resource-operator`
- Wait for successful reconciliation 
- Ensure there are no redis, redissnapshot, postgres, postgressnapshot CRs left in the CRO namespace
- Delete the ROSA STS cluster using the script from delorean: `make -f make/ocm.mk ocm/sts/cluster/delete CLUSTER_NAME=sts-test`

## Checklist
